wheelfile.py

```diff
@@ -570,13 +571,34 @@ class WheelRecord:
     "The .dist-info directory" section of PEP-427, and
     https://packaging.python.org/specifications/recording-installed-packages/.
     """
-    HASH_ALGO = hashlib.sha256
     HASH_BUF_SIZE = 65536
 
     _RecordEntry = namedtuple('_RecordEntry', 'path hash size')
 
-    def __init__(self):
+    def __init__(self, hash_algo: str='sha256'):
         self._records: Dict[str, self._RecordEntry] = {}
+        self._hash_algo = ""
+        self.hash_algo = hash_algo
+
+    @property
+    def hash_algo(self) -> str:
+        """Hash algorithm to use to generate RECORD file entries"""
+        return self._hash_algo
+
+    @hash_algo.setter
+    def hash_algo(self, value: str):
+        # per PEP-376
+        if value not in hashlib.algorithms_guaranteed:
+            raise UnsupportedHashTypeError(f"{value} is not a valid record hash")
+        # per PEP 427
+        if value in ('md5', 'sha1'):
+            raise UnsupportedHashTypeError(f"{value} is a forbidden hash type")
+
+        # PEP-427 says the RECORD hash must be sha256 or better.  Does that mean
+        # hashes such as sha224 are forbidden as well though not specifically
+        # called out?
```

Comment:
I think we'll require clarification from PyPA on this at some point. For now it's not an issue - there's hardly any reason to use anything other than `sha256`. I guess I'll create an issue for it, as a reminder.